app/src/utils/order-book-utils.js

```diff
@@ -94,3 +94,51 @@ export const ordersEquals = (ordersArray1, ordersArray2) => {
         return order.id === ordersArray2[index].id && order.quantity === ordersArray2[index].quantity;
     });
 };
+
+export const processOrderBookWithDeltas = (currentListItems, newAsks, newBids) => {
+    const listItems = currentListItems;
+    let firstBid = 0;
+    let maxQuantity = 0;
+
+    newAsks.forEach(ask => {
+        const { price, orders } = ask;
+        if (orders.length === 0) delete listItems[price]; // Remove price level
+        else if (listItems[price]) {
+            listItems[price].orders = orders; // Modify price level
+        } else {
+            listItems[price] = { // Add price level
+                price,
+                orders,
+                type: 'ask',
+            };
+        }
+    });
+
+    newBids.forEach(bid => {
+        const { price, orders } = bid;
+        if (orders.length === 0) delete listItems[price]; // Remove price level
+        else if (listItems[price]) {
+            listItems[price].orders = orders; // Modify price level
+        } else {
+            listItems[price] = { // Add price level
+                price,
+                orders,
+                type: 'bid',
+            };
+        }
+    });
```

Comment:
since this does the same thing, can it be extracted into a function **that looks like** the following:

```js

const updateListItem = (type, newArray, listItems) => {
   newArray.forEach(askOrBid => {
       const { price, orders } = askOrBid;
        if (orders.length === 0) delete listItems[price]; // Remove price level
        else if (listItems[price]) {
            listItems[price].orders = orders; // Modify price level
        } else {
            listItems[price] = { // Add price level
                price,
                orders,
                type,
            };
      }
```